app/controllers/application_controller.rb

```diff
@@ -25,11 +25,25 @@ def require_login
     redirect_to login_path unless current_user
   end
 
+  class HTMLwithPygments < Redcarpet::Render::HTML
+    def block_code(code, language)
+      sha = Digest::SHA1.hexdigest(code)
+      Rails.cache.fetch ["code", language, sha].join('-') do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.